### PR TITLE
Fix Rails/HelperInstanceVariable 2

### PR DESCRIPTION
**Pull request by macanudo527**
#### What? Why?

- Part of #11482 

This removes the instance variables from 5 helper modules, which violates the [Rails/HelperInstanceVariable cop](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railshelperinstancevariable). This allows for more reusability.

It also removes 3 helper methods that were not being used anywhere in the codebase.

#### What should we test?

This should be a simple refactoring and doesn't need further testing.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/helpers/injection_helper.rb [hunk 1]
```diff
@@ -164,7 +165,6 @@ def default_enterprise_query
   end
 
   def enterprise_injection_data
-    @enterprise_injection_data ||= OpenFoodNetwork::EnterpriseInjectionData.new
-    { data: @enterprise_injection_data }
+    { data: OpenFoodNetwork::EnterpriseInjectionData.new }
```
Comment (mkllnk) [inline app/helpers/injection_helper.rb:157]: This is changing logic. The object was cached before which is important for performance in this case. Caching in helpers isn't great but I'm not sure how easy it is to resolve. We would need to look at all the places where the helpers are called. But really, injecting JSON data into the page is only used by AngularJS which we are phasing out. So keeping it as todo is probably best.

File: app/helpers/shared_helper.rb [hunk 2]
```diff
@@ -3,11 +3,11 @@
 module SharedHelper
   def distributor_link_class(distributor)
     cart = current_order(true)
-    @active_distributors ||= Enterprise.distributors_with_active_order_cycles
+    active_distributors = Enterprise.distributors_with_active_order_cycles
```
Comment (mkllnk) [inline app/helpers/shared_helper.rb:6]: Same here, we need the caching.

File: app/helpers/shop_helper.rb [hunk 3]
```diff
@@ -1,12 +1,12 @@
 # frozen_string_literal: true
 
 module ShopHelper
-  def oc_select_options
-    @order_cycles.map { |oc| { time: pickup_time(oc), id: oc.id } }
+  def oc_select_options(order_cycles)
+    order_cycles.map { |oc| { time: pickup_time(oc), id: oc.id } }
```
Comment (mkllnk) [inline app/helpers/shop_helper.rb:5]: And more here, I won't mark them all.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Thanks, there are some really good changes here. We just need to keep the caching.

**Comment by macanudo527** [comment]
@mkllnk I think I kept all the caching in. Something seems to have gone wrong with the tests though. I'm not getting these errors locally.

**Review by mkllnk — APPROVED** [review]
Great, thanks.

File: app/helpers/injection_helper.rb [hunk 4]
```diff
@@ -105,16 +106,16 @@ def inject_open_street_map_config
...
   end
 
   def inject_available_countries
     inject_json_array "availableCountries", available_countries, Api::CountrySerializer
   end
 
-  def inject_enterprise_attributes
+  def inject_enterprise_attributes(enterprise_attributes)
     render partial: "json/injection_ams",
            locals: { name: 'enterpriseAttributes', json: @enterprise_attributes.to_json.to_s }
```
Comment (rioug) [inline app/helpers/injection_helper.rb]: ~~This one got missed I think :~~ Fixed in the next commit

File: app/views/spree/orders/order_cycle_expired.html.haml [hunk 5]
```diff
@@ -2,7 +2,7 @@
   = t :orders_oc_expired_headline
 
 %p
-  - if active_order_cycle_for_distributor? current_distributor
+  - if active_order_cycle_for_distributor? @distributor
```
Comment (rioug) [inline app/views/spree/orders/order_cycle_expired.html.haml:5]: Why the change from `current_distributor` to `@distributor` ?

Comment (mkllnk) [reply app/views/spree/orders/order_cycle_expired.html.haml:5]: The helper method was ignoring the argument (which was `current_distributor`) and used `@distributor` instead. Now we are passing that on and use the argument.

When looking at an order, we need to display information about the distributor of the order and not of the `current_distributor` which is the shop you selected last. It may be different.

Comment (rioug) [reply app/views/spree/orders/order_cycle_expired.html.haml:5]: I see ! thanks for the clarification.

**Review by rioug — APPROVED** [review]

**Comment by rioug** [comment]
Merging as no manual test needed.
